framework/src/modules/dpos_v2/module.ts

```diff
@@ -131,6 +136,39 @@ export class DPoSModule extends BaseModule {
 		});
 	}
 
+	public metadata(): ModuleMetadata {
+		return {
+			endpoints: [
+				{
+					name: this.endpoint.getAllDelegates.name,
+					response: getAllDelegatesResponseSchema,
+				},
+				{
+					name: this.endpoint.getDelegate.name,
+					request: getDelegateRequestSchema,
+					response: getDelegateResponseSchema,
+				},
+				{
+					name: this.endpoint.getVoter.name,
+					request: getVoterRequestSchema,
+					response: getVoterResponseSchema,
+				},
```

Comment:
I think names can be automated. The thing you use is `Function.prototype.name` right? Then maybe something can iterate over the class properties, pick methods and save us from typing the assignment? Also schema names have a pattern too, if these patterns will be consistent in the future we can entirely create endpoints property from a function.

Reply:
name can be automated, but I couldn't think of any good way to handle request/response schema. Any suggestion?

Reply:
I think it would be better not to automate as it might make things more fragile later on.